Parser/asdl_c.py

```diff
@@ -947,8 +966,26 @@ def visitProduct(self, prod, name):
             fields = name+"_fields"
         else:
             fields = "NULL"
-        self.emit('state->%s_type = make_type("%s", state->AST_type, %s, %d);' %
-                        (name, name, fields, len(prod.fields)), 1)
+        docstring = asdl_of(name, prod)
+        if '\n' in docstring or '(' in docstring:
+            self.emit(
+                'state->%s_type = make_type("%s", state->AST_type, %s, %d,'
+                % (name, name, fields, len(prod.fields)),
+                1,
+                reflow=False,
+            )
+            self.emit(
+                '%s);' % reflow_c_string(docstring, 2),
+                2,
+                reflow=False,
+            )
+        else:
+            self.emit(
+                'state->%s_type = make_type("%s", state->AST_type, %s, %d, %s);'
+                % (name, name, fields, len(prod.fields), reflow_c_string(docstring, 2)),
+                1,
+                reflow=False,
+            )
```

Comment:
My apologies, I sent this comment prematurely. It was a general code written before I noticed that we know the type of the node in every case. In this particular site it may be simpler:
```suggestion
        self.emit(
            'state->%s_type = make_type("%s", state->AST_type, %s, %d,' %
                  (name, name, fields, len(prod.fields)), 1)
        self.emit('%s);' % reflow_c_string(asdl_of(name, prod), 2), 2, reflow=False)
```

Reply:
No problem, this wasn't ready for review though. I did push these to review/edit locally. It is hard to use github's interface for this.